printers/printerTree.php

```diff
@@ -42,15 +42,17 @@ function _orderTree($tree) {
         $this->_sorter->sort($tree->children);
 
         foreach($tree->children as $subTree){
-            $this->_orderTree($subTree);
+            if (is_object($subTree)) {
+                $this->_orderTree($subTree);
+            }
         }
         return $tree;
     }
 
     private function _groupByNs($tab) {
         $tree = new NspagesTreeNsNode(':');
         foreach($tab as $item){
-            $this->_fillTree($tree, $this->_getNS($item), $item, '');
+            $this->_fillTree($tree, $this->_getNS($item), $item, '', ':');
```

Comment:
This is needed for recursion. Without this initial value you will not have correct root of tree.

Reply:
This additional parameter is needed because I remove distinction between pages and children from the tree. Page can have childrens (that means subns, and ns can be a page) and page without childrens is simply page. e.g. if on disk you vill have: pages/tools/proxmox.txt ; pages/tools/proxmox/pbs.txt -> you will have correct tree without duplication entries: one for ns proxmox containing pbs.txt one for proxmox page.
![obraz](https://github.com/user-attachments/assets/5c9ea876-e50b-4b7b-bd93-2663f0d1fd61)